### ISSUE TITLE
Timeout when analyzing

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#173]
### Version

0.4.0

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - id: get-version
           run: echo "Problem!"
         - id: check-for-approved-release
           run: echo "Problem!"
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

Code is analyzed, finding reported.

### Actual behavior

Check does not finish in 2 minutes.

### Additional Context

According to `JVisualVM` thread dump:
```
java.lang.Thread.State: RUNNABLE
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:26)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:19)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:26)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:26)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:25)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.utils.EditDistanceKt.editDistance(SourceFile:24)
        at net.twisterrob.ghlint.rules.DuplicateStepIdRule$visitNormalJob$similar$2.invoke(SourceFile:4)
```
